pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -278,14 +278,20 @@ def should_reset_tensors(self, fx: str) -> bool:
             is_first_batch = bool(self._batch_idx) + self._split_idx == 0
         return is_different_fx and is_first_batch
 
-    def reset(self, metrics: Optional[bool] = None) -> None:
-        if self.trainer.sanity_checking:
+    def reset_metrics(self) -> None:
+        self._progress_bar_metrics = {}
+        self._logged_metrics = {}
+        self._callback_metrics = {}
+
+    def reset(self, metrics: Optional[bool] = None, trainer_metrics: bool = False) -> None:
+        if trainer_metrics:
             # reset metrics
-            self._progress_bar_metrics = {}
-            self._logged_metrics = {}
-            self._callback_metrics = {}
-        assert self.trainer._results is not None
-        self.trainer._results.reset(metrics=metrics)
+            self.reset_metrics()
+
+        if self.trainer.state.fn != TrainerFn.PREDICTING:
```

Comment:
can you explain this check?

Reply:
I can, the prediction loops do not define `_results` (they are `None`)

But i'd change the code to:


```python
if self.trainer._results is not None:
    self.trainer._results.reset(metrics=metrics)
```

Instead of the predict check and the assertion

Reply:
Yeah this is better, Will update this.. thanks @carmocca